cms/static/js/factories/import.js

```diff
@@ -33,39 +52,41 @@ define([
             maxChunkSize: 20 * 1000000, // 20 MB
             autoUpload: false,
             add: function(e, data) {
-                Import.clearImportDisplay();
-                Import.okayToNavigateAway = false;
+                Import.reset();
                 submitBtn.unbind('click');
+
                 file = data.files[0];
+
                 if (file.name.match(/tar\.gz$/)) {
-                    submitBtn.click(function(event){
+                    submitBtn.click(function(event) {
                         event.preventDefault();
-                        $.cookie('lastfileupload', file.name);
+
+                        Import.start(
+                            file.name,
+                            feedbackUrl.replace('fillerName', file.name)
+                        );
+
                         submitBtn.hide();
-                        Import.startUploadFeedback();
-                        data.submit().complete(function(result, textStatus, xhr) {
-                            window.onbeforeunload = null;
-                            if (xhr.status != 200) {
+                        data.submit().complete(function (result, textStatus, xhr) {
+                            if (xhr.status !== 200) {
                                 var serverMsg, errMsg, stage;
+
                                 try{
                                     serverMsg = $.parseJSON(result.responseText);
                                 } catch (e) {
                                     return;
                                 }
-                                errMsg = serverMsg.hasOwnProperty('ErrMsg') ?  serverMsg.ErrMsg : '' ;
+
+                                errMsg = serverMsg.hasOwnProperty('ErrMsg') ? serverMsg.ErrMsg : '' ;
```

Comment:
Oh thanks for spotting it, could you give me the exact steps on how to reproduce it?

Also, related to the errors: I really would like to encapsulate the error strings (the `defaults` array) in the `CourseImport` object, since I think it make more sense for them to belong to it as a private property. The problem is that the `dbError` string (error for the last step) is based on the `library` variable passed to the factory, which is not accessible to `CourseImport` (unless you pass to it at some point, but it would seem too much like an ad-hoc solution to me).

Since `library` is used just to change a word in the error message, would it be possible to make the message more generic so that we don't need it? Or what do you think would be the best approach?

Thanks!